sway/ipc-server.c

```diff
@@ -322,6 +327,46 @@ void ipc_get_pixels(wlc_handle output) {
 	ipc_get_pixel_requests = unhandled;
 }
 
+static int ipc_selection_data_cb(int fd, uint32_t mask, void *data)
+{
+	assert(data);
+	struct get_clipboard_request *req = (struct get_clipboard_request*) data;
+
+	if (mask & WLC_EVENT_ERROR) {
+		sway_log(L_ERROR, "Selection data fd error");
+		const char *error = "{ \"success\": false, \"error\": "
+			"\"Could not receive text data from clipboard\" }";
+		ipc_send_reply(req->client, error, (uint32_t)strlen(error));
+		goto cleanup;
+	}
+
+	if (mask & WLC_EVENT_READABLE) {
+		char buf[512];
+		int ret = read(fd, buf, 511);
+		if (ret < 0) {
+			sway_log_errno(L_ERROR, "Reading from selection data fd failed");
+			const char *error = "{ \"success\": false, \"error\": "
+				"\"Could not receive text data from clipboard\" }";
+			ipc_send_reply(req->client, error, (uint32_t)strlen(error));
+			goto cleanup;
+		}
+
+		buf[ret] = '\0';
+		json_object *obj = json_object_new_object();
+		json_object_object_add(obj, "success", json_object_new_boolean(true));
+		json_object_object_add(obj, "content", json_object_new_string(buf));
```

Comment:
ok. yes, wildcards sound possible and are a good idea to not query for "text/plain" and "text/plain;charset=utf-8" every time.